### PR TITLE
OSX installer prototype with pyinstaller base

### LINKED ISSUE #962: OSX Installer using pyinstaller

### PR DESCRIPTION
**Pull request by ilia-tsyplenkov**
Fixes #962 

This is OSX installer prototype.
Since we can't to downgrade/upgrade OS python the main idea to install required (for mindsdb correct work) python version in a custom directory (configure, compile and install). Next steps are similar with windows installer (install torch in specified version range, create a primitive mindsdb launcher).
The installer also checks that require programs/packages/utils installed and stops work with related message otherwise.

### CODE REVIEW

**Review by George3d6 — CHANGES_REQUESTED** [review]
Building the installer should also be added to the CI (`.github/workflows/mindsdb.yml` and `release.yml`)

**Comment by ilia-tsyplenkov** [comment]
> Building the installer should also be added to the CI (`.github/workflows/mindsdb.yml` and `release.yml`)

Sure, but I decided that general idea needs to be discussed first. Going to add required changes.

File: .github/workflows/mindsdb.yml [hunk 1]
```diff
@@ -172,10 +204,6 @@ jobs:
         SOURCE_DIR: 'distributions/linux/dist'
         DEST_DIR: 'mindsdb-installer/linux'
 
-    - name: Build OSX sh installer
-      run: |
-        python distributions/osx/rename_installers.py beta
```
Comment (George3d6) [inline .github/workflows/mindsdb.yml:177]: Don't remove this, we can still keep the script installer.

**Review by George3d6 — CHANGES_REQUESTED** [review]

File: .github/workflows/mindsdb.yml (hunk 1 above)
Comment (ilia-tsyplenkov) [reply .github/workflows/mindsdb.yml:181]: Done

**Pull request merged**